mac-x64 compilation fpusup.s:140:20: error: unexpected token in '.section' directive .section .note.GNU-stack,"",@progbits

Failure link
------------

https://openj9-jenkins.osuosl.org/job/Build_JDKnext_x86-64_mac_OpenJDK/787/console

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
23:31:57  [ 22%] Building ASM object runtime/util/CMakeFiles/j9util.dir/fpusup.s.o
23:31:57  /Users/jenkins/workspace/Build_JDKnext_x86-64_mac_OpenJDK/build/macosx-x86_64-server-release/vm/runtime/util/fpusup.s:140:20: error: unexpected token in '.section' directive
23:31:57   .section .note.GNU-stack,"",@progbits
23:31:57                     ^
23:31:57  make[6]: *** [runtime/util/CMakeFiles/j9util.dir/fpusup.s.o] Error 1
23:31:57  make[6]: *** Waiting for unfinished jobs....
23:31:57  make[5]: *** [runtime/util/CMakeFiles/j9util.dir/all] Error 2
23:31:57  make[4]: *** [all] Error 2
23:31:57  make[3]: *** [build-j9] Error 2
23:31:57  make[2]: *** [j9vm-build] Error 2
```

It occurred across Java levels [jdk23-mac-x64](https://hyc-runtimes-jenkins.swg-devops.com/job/build-scripts/job/jobs/job/jdk23/job/jdk23-mac-x64-openj9/48/console), [jdk21u-mac-x64](https://hyc-runtimes-jenkins.swg-devops.com/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-mac-x64-openj9-IBM/272/console), [jdk17u-mac-x64](https://hyc-runtimes-jenkins.swg-devops.com/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-mac-x64-openj9-IBM/783/console), and [jdk11u-mac-x64](https://hyc-runtimes-jenkins.swg-devops.com/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-mac-x64-openj9-IBM/934/console)


It seems related to
* https://github.com/eclipse-openj9/openj9/pull/20278

FYI @keithc-ca

--- a/runtime/oti/xhelpers.m4
+++ b/runtime/oti/xhelpers.m4
@@ -110,9 +110,9 @@ define({START_PROC},{
 	GLOBAL_SYMBOL($1):
 })
 
-define({FILE_END},{
+define({FILE_END},{ifdef({LINUX},{
 	.section .note.GNU-stack,"",@progbits
-})
+})})
 
 define({END_PROC},{
 END_$1:

